drivers/include/sx127x.h

```diff
@@ -0,0 +1,703 @@
+    SX127X_CR_4_5 = 1,                 /**< coding rate 4/5 */
+    SX127X_CR_4_6,                     /**< coding rate 4/6 */
+    SX127X_CR_4_7,                     /**< coding rate 4/7 */
+    SX127X_CR_4_8                      /**< coding rate 4/8 */
+};
+
+/**
+ * @brief   Radio driver internal state machine states definition.
+ */
+enum {
+    SX127X_RF_IDLE = 0,                /**< Idle state */
+    SX127X_RF_RX_RUNNING,              /**< Sending state */
+    SX127X_RF_TX_RUNNING,              /**< Receiving state */
+    SX127X_RF_CAD,                     /**< Channel activity detection state */
+};
+
+/**
+ * @brief   Event types.
+ */
+enum {
+    SX127X_RX_DONE = 0,                /**< Receiving complete */
+    SX127X_TX_DONE,                    /**< Sending complete*/
+    SX127X_RX_TIMEOUT,                 /**< Receiving timeout */
+    SX127X_TX_TIMEOUT,                 /**< Sending timeout */
+    SX127X_RX_ERROR_CRC,               /**< Receiving CRC error */
+    SX127X_FHSS_CHANGE_CHANNEL,        /**< Channel change */
+    SX127X_CAD_DONE,                   /**< Channel activity detection complete */
+};
+
+/**
+ * @brief   LoRa configuration structure.
+ */
+typedef struct {
+    uint8_t power;                     /**< Signal power */
+    uint8_t bandwidth;                 /**< Signal bandwidth */
+    uint8_t datarate;                  /**< Spreading factor rate, e.g datarate */
+    bool low_datarate_optimize;        /**< Optimize for low data rate */
+    uint8_t coderate;                  /**< Error coding rate */
+    uint16_t preamble_len;             /**< Length of preamble header */
+    bool use_fix_len;                  /**< Use fixed header len, (enable implicit header mode) */
+    bool crc_on;                       /**< Enable payload CRC */
+    bool freq_hop_on;                  /**< Enable frequency hopping */
+    uint8_t freq_hop_period;           /**< Frequency hop period */
+    bool iq_inverted;                  /**< Set inverted IQ */
+    bool rx_continuous;                /**< Use continuous reception */
+    uint32_t rx_timeout;               /**< RX timeout in symbols */
+    uint32_t tx_timeout;               /**< TX timeout in symbols */
+} sx127x_lora_settings_t;
+
+/**
+ * @brief   Radio settings.
+ */
+typedef struct {
+    uint8_t state;                     /**< Radio state */
+    uint8_t modem;                     /**< Driver model (FSK or LoRa) */
+    uint32_t channel;                  /**< Radio channel */
+    sx127x_lora_settings_t lora;       /**< LoRa settings */
+    uint32_t window_timeout;           /**< Timeout window */
+} sx127x_radio_settings_t;
+
+/**
+ * @brief   SX127X internal data.
+ */
+typedef struct {
+    /* Data that will be passed to events handler in application */
+    uint32_t last_channel;             /**< Last channel in frequency hopping sequence */
+    bool is_last_cad_success;          /**< Sign of success of last CAD operation (activity detected) */
```

Comment:
This bool can stay, since the we can not win any RAM by re-packing. The two xtimers should go to the top of the struct however.